Foundation/Conduit/Textual.hs

```diff
@@ -0,0 +1,46 @@
+module Foundation.Conduit.Textual
+    ( lines
+    , fromBytes
+    , toBytes
+    ) where
+
+import           Foundation.Internal.Base hiding (throw)
+import           Foundation.Array.Unboxed
+import           Foundation.String (String)
+import qualified Foundation.String.UTF8 as S
+import           Foundation.Conduit.Internal
+import           Foundation.Monad
+
+-- | Split conduit of string to its lines
+--
+-- This is very similar to Prelude lines except
+-- it work directly on Conduit
+lines :: Monad m => Conduit String String m ()
+lines = await >>= maybe (finish mempty) (go mempty)
+  where
+    finish buf = if S.null buf then return () else yield buf
+
+    go current nextBuf =
+        case S.uncons next' of
+            Just (_, rest') -> yield (current `mappend` line) >> go mempty rest'
+            Nothing         ->
+                let nextCurrent = current `mappend` nextBuf
```

Comment:
This is going to be _O(n^2)_. I'd recommend a list of buffers and doing a reverse at the end (or a builder, if you want to be fancy). I had to submit a patch to fix a bug like this in the original conduit since it can easily come up (reading a huge file with no lines).